### ISSUE TITLE
`set!` for `ECCOMetadata` will inpaint the whole ocean even for a single column grid

### ISSUE THREAD

**Issue by glwagner** [CliMA/ClimaOcean.jl#174]
It would make sense to restrict the size of the field to fit to the grid in question I think. Here's the code:

https://github.com/CliMA/ClimaOcean.jl/blob/b3b44b709c70b296adb63cf7b2f4552334dcfd1b/src/DataWrangling/ECCO.jl#L295-L303

cc @simone-silvestri

**Comment by simone-silvestri** [comment]
Yep, it does at the moment. We can optimize this step for single columns if we know the column is in a valid location.

**Comment by glwagner** [comment]
Can't we optimize for any regional model? Just cut out the unneeded areas before inpainting.
